gcc/testsuite/g++.dg/contracts/cpp26/noipa.C

```diff
@@ -0,0 +1,21 @@
+// check that we do not optimise based on contract_asssert
+// { dg-options "-O3 -std=c++23 -fcontracts -fcontracts-nonattr -fcontract-evaluation-semantic=quick_enforce -fdump-tree-optimized " }
+// { dg-do compile }
+
+[[gnu::used, gnu::noinline]]
+inline int* f(int *y) {
+  contract_assert(y != nullptr);
+  return y;
+}
+
+
+int foo(int *x) {
+    if (f(x) == nullptr)
+    {
+        return 0;
+    }
+    return *x;
+}
+
+// Check that the if condition in foo isn't optimised away
+// { dg-final { scan-tree-dump "if \\(_. == 0B\\)" "optimized" } }
```

Comment:
I suspect that this is somewhat fragile - although both macOS and Linux generate similar code here, not sure that would apply universally.

maybe something like

```
scan-tree-dump-times {if \(.* == 0B\)} 2 "optimized" }
```
(untested - but the idea is to require that there are two check not one)